src/main/java/com/nike/cerberus/operation/core/UpdateStackTagsOperation.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2018 Nike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.operation.core;
+
+import com.amazonaws.AmazonServiceException;
+import com.nike.cerberus.command.core.UpdateStackTagsCommand;
+import com.nike.cerberus.domain.environment.Stack;
+import com.nike.cerberus.operation.Operation;
+import com.nike.cerberus.service.CloudFormationService;
+import com.nike.cerberus.store.ConfigStore;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Map;
+
+import static com.nike.cerberus.module.CerberusModule.ENV_NAME;
+
+/**
+ * Operation for updating stack tags.
+ */
+public class UpdateStackTagsOperation implements Operation<UpdateStackTagsCommand> {
+
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final CloudFormationService cloudFormationService;
+    private final String environmentName;
+    private final ConfigStore configStore;
+
+    @Inject
+    public UpdateStackTagsOperation(CloudFormationService cloudFormationService,
+                                    @Named(ENV_NAME) String environmentName,
+                                    ConfigStore configStore) {
+
+        this.cloudFormationService = cloudFormationService;
+        this.environmentName = environmentName;
+        this.configStore = configStore;
+    }
+
+    @Override
+    public void run(UpdateStackTagsCommand command) {
+        String stackId = command.getStack().getFullName(environmentName);
+        Map<String, String> parameters = cloudFormationService.getStackParameters(configStore.getPrimaryRegion(), stackId);
+        Map<String, String> tags = command.getTagsDelegate().getTags();
+        if (!command.isOverwriteTags()){
+            Map<String, String> existingTags = cloudFormationService.getStackTags(configStore.getPrimaryRegion(), stackId);
+            existingTags.forEach((k, v) -> tags.merge(k, v, (o, n)->o));
+        }
+        if (Stack.DATABASE.equals(command.getStack())) {
```

Comment:
Why do you need this, this command should not be updating existing stack parameters, but rather using the template and parameters that already exist.

Reply:
The parameters are required for updating stacks or creating change sets. Usually passing in the existing parameters works well. But for the database stack, the database password comes back as asterisks. That's why I need to pull the real value from the configStore.